src/Microsoft.ML.Data/Transforms/HashTransform.cs

```diff
@@ -188,109 +199,65 @@ private static VersionInfo GetVersionInfo()
                 loaderSignature: LoaderSignature);
         }
 
-        private readonly ColInfoEx[] _exes;
-        private readonly ColumnType[] _types;
-
+        private readonly ColumnInfo[] _columns;
         private readonly VBuffer<DvText>[] _keyValues;
         private readonly ColumnType[] _kvTypes;
 
-        public static HashTransform Create(IHostEnvironment env, ModelLoadContext ctx, IDataView input)
+        protected override void CheckInputColumn(ISchema inputSchema, int col, int srcCol)
         {
-            Contracts.CheckValue(env, nameof(env));
-            var h = env.Register(RegistrationName);
-            h.CheckValue(ctx, nameof(ctx));
-            h.CheckValue(input, nameof(input));
-            ctx.CheckAtModel(GetVersionInfo());
-            return h.Apply("Loading Model", ch => new HashTransform(h, ctx, input));
-        }
-
-        private HashTransform(IHost host, ModelLoadContext ctx, IDataView input)
-            : base(host, ctx, input, TestType)
-        {
-            Host.AssertValue(ctx);
-
-            // *** Binary format ***
-            // <base>
-            // Exes
-
-            Host.AssertNonEmpty(Infos);
-            _exes = new ColInfoEx[Infos.Length];
-            for (int iinfo = 0; iinfo < Infos.Length; iinfo++)
-                _exes[iinfo] = new ColInfoEx(ctx);
-
-            _types = InitColumnTypes();
-
-            TextModelHelper.LoadAll(Host, ctx, Infos.Length, out _keyValues, out _kvTypes);
-            SetMetadata();
+            var type = inputSchema.GetColumnType(srcCol);
+            string reason = TestType(type);
```

Comment:
>TestType [](start = 28, length = 8)

In this case, I think we should make it just part of `CheckInputColumn` and throw `ExceptSchemaMismatch`.